mohawk/util.py

```diff
@@ -268,3 +267,28 @@ def normalize_header_attr(val):
     # Normalize like the hawk reference code.
     val = escape_header_attr(val)
     return val
+
+
+def get_bewit(resource):
+    """Returns a bewit identifier for the resource as a string"""
+    assert resource.method == 'GET'
+    mac = calculate_mac(
+        'bewit',
+        resource,
+        None,
+    )
+
+    if isinstance(mac, six.binary_type):
+        mac = mac.decode('ascii')
+
+    if resource.ext is None:
+        ext = ''
+    else:
+        ext = resource.ext
+
+    inner_bewit = u"{id}\\{exp}\\{mac}\\{ext}".format(
+        id=resource.credentials['id'],
+        exp=resource.timestamp,
+        ext=ext,
+        mac=mac).encode('ascii')
+    return urlsafe_b64encode(inner_bewit).decode("ascii")
```

Comment:
why are you decoding the unicode object to ascii? Did you mean to encode it? Encoding it would return bytes. Your code here will take a unicode object, encode it with the default system encoding, and decode it back into unicode.